src/impl_methods.rs

```diff
@@ -2131,6 +2131,20 @@ where
         }
     }
 
+    /// Flatten the array view to a one-dimensional array.
+    ///
+    /// ```
+    /// use ndarray::{arr1, arr3};
+    ///
+    /// let array = arr3(&[[[1, 2], [3, 4]], [[5, 6], [7, 8]]]);
+    /// let flattened = array.flatten();
+    /// assert_eq!(flattened, arr1(&[1, 2, 3, 4, 5, 6, 7, 8]));
+    /// ```
+    pub fn flatten(self) -> ArrayBase<S, Ix1> {
+        let len = self.len();
+        self.into_shape_with_order((len,)).unwrap()
```

Comment:
will panic on some arrays that are not contiguous. Another design is needed unfortunately.

Reply:
Suggestion: base it on to_shape and return the same kind of array that it does. Have both flatten and flatten with order to handle the optional order argument.

Reply:
@bluss patched, what do you think?